### ISSUE TITLE
Default to user installation

### ISSUE THREAD

**Issue by adrm** [flatpak/flatpak#4099]
## Linux distribution and version
All

## Flatpak version
All

## Description of the problem
As a normal home user of flatpak, it doesn't makes sense to me that all flatpak commands are run at the system level by default. In a shared computer, or single-user computer, there is no need for system-wide installations or updates.

Since as far as I understand the private user is the target user of flatpak, it seems to me that it is a saner default that flatpaks are installed on the user level so that:
- Unexpected apps do not appear for other users in shared multi-user systems.
- Installation might be faster if it does not need system privileges.
- The expectations of the typical user match the defaults of the tool: when I install an app it is installed in my user account.

I am suggesting this from the perspective of an user, so there might be other reasons for this default which I don't know, but this kind of defaults trickle down to final-user tools like Gnome Software and others, and that is why I think it is important to set the best possible defaults.

**Comment by AdrianVovk** [comment]
I agree with this sentiment, especially with the advent of systemd-homed. However, there's always the problem of compatibility. Tools expect Flatpak to default to the system installation. Maybe this can happen at a 2.0 or a big API/ABI break? Also, end-user tools like GNOME Software or Souk can use the user installation on their own terms without Flatpak defaulting to it. That's just not implemented, and in fact user-installed Flatpaks can trip these tools up because they don't expect it (which, IMO, is really bad. I have issues open on these tools to get them to support/default to user installs)

**Comment by mwleeds** [comment]
I think the logic for defaulting to installing apps system-wide is that is how apps are installed on package-based systems (deb/rpm) and therefore the expected behavior. See the discussion here: https://gitlab.freedesktop.org/pwithnall/malcontent/-/issues/30#note_687925

**Comment by LeonardoSanBenitez** [comment]
I also feel that user-wise installation should be the default

**Comment by Rcomian** [comment]
Just been searching google to find out how to make flatpak use the --user flag by default. I was expecting to find a configuration file somewhere.

Adding this comment to support this issue tho.

**Comment by mwleeds** [comment]
If you want Flatpak to default to installing to your user installation, you can simply only configure remotes in your user installation. The Flatpak CLI will then use the user installation without requiring you specify `--user`.

That is assuming you have control over the whole system. If it's a case where the sys admin has configured remotes in the system-wide installation and you want to have your own Flatpaks in the per-user one, I can understand why you'd want to be able to default to per-user rather than system-wide. But (a) there's a [reason](https://gitlab.freedesktop.org/pwithnall/malcontent/-/issues/30#note_687925) Flatpak defaults to system-wide, (b) the idea of a config file has been discussed and rejected in #2887, (c) it's not possible to change the default without breaking backwards compatibility, and (d) multi-user systems where you have a sys admin controlling the system-wide installation are probably rare anyway.

So I don't really see this issue going anywhere...

**Comment by akdev1l** [comment]
Hi, I've been running an OSTree system build using OCI for a while.

With a move towards immutability on the desktop as represented by Fedora Silverblue picking up in popularity and projects like [uBlue](https://ublue.it) (disclaimer: I'm a contributor) coming out I think it would be a good time to reset this default to a more saner (immutable-er) default. 

I've been running flatpak --user for years and I have never run into any issues so it seems quality-wise the software is quite ready.

**Comment by johnmmcgee** [comment]
I would like the option to default to user, without the need to disable the system wide flatpak repos.  I state this as it allows for you to allow some apps to stay system wide and allow per user apps as well.

**Comment by pkern** [comment]
https://docs.flathub.org/docs/for-users/user-vs-system-install/ suggests that per-user installation is recommended for most cases but then it's not possible to switch over the default for the CLI. That feels really weird. Similarly I was really surprised about the polkit rule just doing things without further auth to the system store. Yes, most of us run single user systems anyway. But I feel like there's not a consistent story. And the reason I ran into this is that VS Codium then suggests in its documentation to modify various bits of the sandbox for further isolation, but *does* include `--user`, because it suggests inherently per-user settings. At least then the two settings buckets should be merged? Instead the defaults will cause the modifications to be silently dropped unless you really know what you are doing.

I admit, I have a lot of trouble understanding the intent of the UX complexity here.

**Comment by smcv** [comment]
`docs.flathub.org` is the separate project https://github.com/flatpak/flatpak-docs/ and not all of it is written or reviewed by the same people who are actually writing the code. I would probably not have accepted the merge request that added that text, if I had been the reviewer (but I wasn't).

I didn't do the early design of Flatpak or choose to default to `--system`, so please don't shoot the messenger.

> it's not possible to switch over the default for the CLI

Probably not, no. Regardless of whether, with hindsight, `--user` might have been a better default or not, changing the default now would be a backwards-compatibility break.

> VS Codium then suggests in its documentation to modify various bits of the sandbox for further isolation, but does include --user, because it suggests inherently per-user settings

If its documentation is discussing `flatpak override`, then, yes, the same `--system` and `--user` options don't mean exactly the same things there that they would for `flatpak install` and similar. `flatpak override [--system]` is a privileged operation and sets up overrides that apply for all users, whereas `flatpak override --user` is an unprivileged operation and sets up higher-precedence overrides that apply to the current user only. This is orthogonal to whether the app itself is installed system-wide or per-user.

There is a multi-layer stack of sandbox parameters (mainly access-related like `--allow` and `--filesystem`, but also non-access-related things like `--env`), where the highest layer that specifies a particular parameter takes priority over lower layers. From memory, it goes something like this:

* highest priority: `flatpak run` command-line options
* `flatpak override --user` with an app ID
* `flatpak override --user` without an app ID (applies to all apps)
* `flatpak override [--system]` with an app ID (applies to all users who run that app)
* `flatpak override [--system]` without an app ID (applies to all apps and all users)
* the app's manifest
* lowest priority: the runtime's manifest

The `flatpak override` man page could probably be clearer about this (merge requests welcome).
